npm_and_yarn/lib/dependabot/npm_and_yarn/file_parser.rb

```diff
@@ -112,6 +113,15 @@ def lockfiles
         }
       end
 
+      sig { returns(T::Hash[Symbol, T.nilable(Dependabot::DependencyFile)]) }
+      def rcfiles
+        {
+          npm: npmrc,
```

Comment:
should these be the constants?

Reply:
Those are methods defined later in the code

Reply:
We didn't have rcfiles as passing parameters. That's why I created this. file_fetcher and file_parser both needs to get registery and set that to core pack registry environment variable.